user/super/com/google/gwt/emul/java/lang/Character.java

```diff
@@ -234,57 +225,145 @@ public static boolean isBmpCodePoint(int codePoint) {
     return codePoint >= MIN_VALUE && codePoint <= MAX_VALUE;
   }
 
+  private static NativeRegExp definedRegex;
+
+  public static boolean isDefined(char c) {
+    return isDefined(String.valueOf(c));
+  }
+
+  public static boolean isDefined(int codePoint) {
+    return isValidCodePoint(codePoint)
+        && isDefined(String.NativeString.fromCodePoint(codePoint));
+  }
+
+  private static boolean isDefined(String str) {
+    if (definedRegex == null) {
+      definedRegex = new NativeRegExp("\\P{Cn}", "u");
+    }
+    return definedRegex.test(str);
+  }
+
   private static NativeRegExp digitRegex;
 
-  /*
-   * TODO: correct Unicode handling.
-   */
   public static boolean isDigit(char c) {
+    return isDigit(String.valueOf(c));
+  }
+
+  // Known differences between Java 17 and Chrome 135
+  // 11f50 .. 11f59, 16ac0 .. 16ac9, 1e4f0 .. 1e4f9, 1fbf0 .. 1fbf9
```

Comment:
Should we leave this method out if it is still wrong? Which side do we consider to be "wrong" here, Java or Chrome? Would it make sense to have a test (possibly ignored) that shows this, so it is easier to reevaluate later?

Also, consider using hex in digit() above, so that it is the same convention as here, or change this to be decimal?